### ISSUE TITLE
GraphQL node: ERROR: Using variables failed

### ISSUE THREAD

**Issue by frane** [n8n-io/n8n#1197]
**Describe the bug**
When using a JSON format query in the GraphQL node, the node fails with `ERROR: Using variables failed: With error message: SyntaxError: Unexpected end of JSON input`. The issue occurs if "Response Format" is JSON and "Variables" is empty. Setting "Variables" to `{}` instead of empty fixes the issue.

**To Reproduce**
![Screen Shot 2020-11-20 at 12 47 15](https://user-images.githubusercontent.com/35420/99798030-8e414980-2b30-11eb-9085-15d4cf4554ae.png)

**Expected behavior**
Should not fail on empty Variables, as other clients don't either.

**Environment (please complete the following information):**
 - OS: Docker
 - n8n Version: 0.94.0
 - Node.js Version: 12

**Additional context**
Seems the change was introduced here: https://github.com/n8n-io/n8n/commit/d9a7d5fa34fe47dc846b3d517a137eda666bd6c8
IMO it's problematic in two ways: 
1. It's a breaking change that drastically changes the behavior of the node, but wan't even flagged as such. It quite literally broke our workflows.
2. It should not disallow empty variables, as this is a quite common situation and other clients also handle it gracefully, i.e. don't impose any restrictions on empty variables.

While I do understand the need to protect against malformed JSON strings, the common case of empty/no variables needs to be considered too.

Suggestion:
In GraphQL.node.ts line 243 change:
```
requestOptions.body.variables = JSON.parse(requestOptions.body.variables);
```
to 
```
requestOptions.body.variables = JSON.parse(requestOptions.body.variables || "{}");
```
That would cover both concerns.

**Comment by janober** [comment]
Ah very sorry. Was not supposed to be a breaking change, in the meaning that it was not supposed to break for empty variables. I was however aware that it would now behave differently and break for people that have invalid variables defined and that was on purpose. Because before it was already broken for those people in the way that it pretended to send data even though it did not. I prefer much more that the workflow fails in that case and people can then fix it.

Got fixed. Will try to push out a bug fix version very soon.

**Comment by janober** [comment]
Fix got released with n8n@0.94.1
